lib/plugins/aws/deploy/compile/events/schedule/index.js

```diff
@@ -38,9 +40,13 @@ class AwsCompileScheduledEvents {
               }
               ScheduleExpression = event.schedule.rate;
               State = event.schedule.enabled ? 'ENABLED' : 'DISABLED';
+              Input = event.schedule.input || '';
```

Comment:
After this line here we can add the logic to parse the input and detect if it's an object (from yaml syntax) or a string (in which case we can assume it's already in JSON format).

Something like this should make the trick:

```javascript
if (typeof Input === 'object') {
  Input = JSON.stringify(Input);
}
```